encoding/thrift/inbound_nowire_test.go

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/thriftrw/protocol/binary"
+	"go.uber.org/thriftrw/protocol/stream"
+	streamtest "go.uber.org/thriftrw/thrifttest/stream"
+	"go.uber.org/thriftrw/wire"
+	"go.uber.org/yarpc/api/transport/transporttest"
+	"go.uber.org/yarpc/internal/testtime"
+)
+
+const _body = "decoded"
+
+type bodyReader struct {
+	body string
+	err  error
+}
+
+func (br *bodyReader) Decode(sr stream.Reader) error {
+	br.body = _body
+	return br.err
+}
+
+type responseEnveloper struct {
+	name         string
+	envelopeType wire.EnvelopeType
+}
+
+var _ stream.Enveloper = (*responseEnveloper)(nil)
+
+func (re responseEnveloper) MethodName() string              { return re.name }
+func (re responseEnveloper) EnvelopeType() wire.EnvelopeType { return re.envelopeType }
+func (re responseEnveloper) Encode(stream.Writer) error      { return nil }
+
+type responseWriter struct{ err error }
+
+var _ stream.ResponseWriter = (*responseWriter)(nil)
+
+func (rw responseWriter) WriteResponse(wire.EnvelopeType, io.Writer, stream.Enveloper) error {
+	return rw.err
+}
+
+type responseHandler struct {
+	t   *testing.T
+	nwc *NoWireCall
+
+	reqBody stream.BodyReader
+	body    stream.Enveloper
+}
+
+var _ NoWireHandler = (*responseHandler)(nil)
```

Comment:
Is there any chance to use a mock for NoWireHandler?

Reply:
In the case of the `NoWireHandler` stub, I wanted to ensure that the stuff thats passed into `HandleNoWire` is what is expected.  Since all of the calls here have effectively the same checks, it seems cleaner to have the stub that does all of this at once instead of repeating these checks for each of the mocks.